TrackletAlgorithm/MatchProcessor.h

```diff
@@ -721,56 +1395,51 @@ void MatchProcessor(BXType bx,
       //can not get to here on first cycle, but compile don't seem to realize 
       //this and fail to reach II=1
       else if (istep != 0) meu.step(instubdata.getMem(iMEU));
-      
-      meu.processPipeLine(table[iMEU]);      
+
+      meu.processPipeLine(table[iMEU]);
 
     } //end MEU loop
     
-    if (hasMatch) {
+    if(hasMatch) {
  
       ap_uint<VMStubMECMBase<VMSMEType>::kVMSMEIDSize> stubindex;
       ap_uint<AllProjection<APTYPE>::kAllProjectionSize> allprojdata;
       
       (stubindex,allprojdata) = matches[bestiMEU];
-      //matchengine[bestiMEU].advance();
-
       AllProjection<APTYPE> allproj(allprojdata);
 
       auto trkindex=(allproj.getTCID(), allproj.getTrackletIndex());
     
       ap_uint<1> newtracklet = lastTrkID != trkindex;
+      isMatch = newtracklet ? ap_uint<1>(0) : isMatch;
       
       lastTrkID = trkindex;
 
       MatchCalculator<ASTYPE, APTYPE, VMSMEType, FMTYPE, maxFullMatchCopies, LAYER, PHISEC>
-	(bx, newtracklet, savedMatch, best_delta_phi, allstub, allproj, stubindex,
-	 nmcout1, nmcout2, nmcout3, nmcout4, nmcout5, nmcout6, nmcout7, nmcout8,
-	 fullmatch);
+        (bx, newtracklet, isMatch, savedMatch, best_delta_z, best_delta_phi, best_delta_rphi, best_delta_r, allstub, allproj, stubindex,
+         nmcout1, nmcout2, nmcout3, nmcout4, nmcout5, nmcout6, nmcout7, nmcout8,
+         fullmatch);
     } //end MC if
-    if (validproj__) {
-
-      projbufferarray.saveProjection(projbuffer__);
-      projbufferarray.incProjection();
-     
-    }
+    
 
       
     if (validin_) {
       auto iphiproj = projdata_.getPhi();
       auto izproj = projdata_.getZ();
       auto iphider = projdata_.getPhiDer();
+      auto izder = projdata_.getRZDer();
       auto trackletid = projdata_.getTCID();
       
       // PS seed
       // top 3 bits of tracklet index indicate the seeding pair
-      ap_uint<nbits_seed> iseed = trackletid.range(trackletid.length()-1,trackletid.length()-nbits_seed);
+      const ap_uint<nbits_seed> iseed = trackletid.range(trackletid.length()-1,trackletid.length()-nbits_seed);
       // Cf. https://github.com/cms-tracklet/fpga_emulation_longVM/blob/fw_synch/FPGATrackletCalculator.hh#L166
       // and here?
       // https://github.com/cms-tracklet/fpga_emulation_longVM/blob/fw_synch/FPGATracklet.hh#L1621
       // NOTE: emulation fw_synch branch does not include L2L3 seeding; the master branch does
       
       // All seeding pairs are PS modules except L3L4 and L5L6
-      bool psseed = not(iseed==TF::L3L4 or iseed==TF::L5L6); 
+      const bool psseed = not(iseed==TF::L3L4 or iseed==TF::L5L6); 
       
       
       // vmproj index
```

Comment:
As far as I can tell nallproj is initialized to zero and never modified? Do we need nallproj? Is index actually used?

Reply:
I've remove the unused variables